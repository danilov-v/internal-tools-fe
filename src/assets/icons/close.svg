<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<svg xmlns="http://www.w3.org/2000/svg" width="9.354" height="10.096" viewBox="0 0 2.475 2.671">
  <g fill="#f2f2f2" stroke="#f2f2f2" stroke-linecap="round">
    <path d="M.14 2.532L2.334.14" stroke-width=".276"/>
    <path d="M2.338 2.493L.138.178" stroke-width=".272"/>
  </g>
</svg>
