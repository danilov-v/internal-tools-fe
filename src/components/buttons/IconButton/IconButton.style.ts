import styled from 'styled-components';

type ContainerProps = {
  color: 'default' | 'primary' | 'secondary';
  disabled: boolean;
  variant: 'default' | 'outlined';
};

enum COLORS {
  default = 'primaryGray',
  primary = 'primaryGreen',
  secondary = 'primaryRed',
}

export const Container = styled.button<ContainerProps>`
  align-items: center;
  background-color: transparent;
  border-radius: 100%;
  border: ${({ color, theme, variant }) =>
    variant === 'outlined' ? `solid 1px ${theme[COLORS[color]]}` : 'none'};
  cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')};
  display: flex;
  height: 24px;
  justify-content: center;
  margin: 4px;
  padding: 4px;
  width: 24px;

  svg {
    height: 100%;
    width: 100%;
  }

  &:hover {
    opacity: ${({ disabled }) => (disabled ? 0.6 : 0.8)};
  }
`;
