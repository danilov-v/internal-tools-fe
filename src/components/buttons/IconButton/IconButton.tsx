import React, { MouseEvent, ReactNode } from 'react';

import * as S from './IconButton.style';

const IconButton: React.FC<IconButtonProps> = ({
  className = '',
  color = 'default',
  disabled = false,
  icon = null,
  name = '',
  onClick = () => {},
  type = 'button',
  variant = 'default',
}) => (
  <S.Container
    className={className}
    color={color}
    disabled={disabled}
    name={name}
    onClick={onClick}
    type={type}
    variant={variant}
  >
    {icon}
  </S.Container>
);

type IconButtonProps = {
  className?: string;
  color?: 'default' | 'primary' | 'secondary';
  disabled?: boolean;
  icon: ReactNode;
  name?: string;
  onClick?: (event: MouseEvent<HTMLButtonElement>) => void;
  type?: 'button' | 'submit' | 'reset';
  variant?: 'default' | 'outlined';
};

export { IconButton };
