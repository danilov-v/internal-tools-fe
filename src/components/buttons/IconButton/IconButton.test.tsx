import React from 'react';
import { render, RenderResult } from '@testing-library/react';

import 'jest-styled-components';

import { ReactComponent as Icon } from 'assets/icons/close.svg';
import { IconButton } from './IconButton';

const getComponent = (props = {}): RenderResult => {
  const parsedProps = {
    icon: <Icon />,
    ...props,
  };

  return render(<IconButton {...parsedProps} />);
};

describe('IconButton component', () => {
  it('matches snapshot with only required props', () => {
    expect(getComponent().baseElement).toMatchSnapshot();
  });

  it('matches snapshots with additional props', () => {
    const defaultPrimaryButton = getComponent({
      color: 'primary',
      disabled: true,
      type: 'button',
      variant: 'default',
    });
    const outlinedSecondaryButton = getComponent({
      color: 'secondary',
      name: 'name',
      type: 'submit',
      variant: 'outlined',
    });

    expect(defaultPrimaryButton.baseElement).toMatchSnapshot();
    expect(outlinedSecondaryButton.baseElement).toMatchSnapshot();
  });
});
