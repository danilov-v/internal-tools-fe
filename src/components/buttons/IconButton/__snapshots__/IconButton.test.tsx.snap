// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton component matches snapshot with only required props 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border-radius: 100%;
  border: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 4px;
  padding: 4px;
  width: 24px;
}

.c0 svg {
  height: 100%;
  width: 100%;
}

.c0:hover {
  opacity: 0.8;
}

<body>
  <div>
    <button
      class="c0"
      color="default"
      name=""
      type="button"
    >
      <svg>
        close.svg
      </svg>
    </button>
  </div>
</body>
`;

exports[`IconButton component matches snapshots with additional props 1`] = `
<body>
  .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border-radius: 100%;
  border: none;
  cursor: default;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 4px;
  padding: 4px;
  width: 24px;
}

.c0 svg {
  height: 100%;
  width: 100%;
}

.c0:hover {
  opacity: 0.6;
}

<div>
    <button
      class="c0"
      color="primary"
      disabled=""
      name=""
      type="button"
    >
      <svg>
        close.svg
      </svg>
    </button>
  </div>
  .c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: transparent;
  border-radius: 100%;
  border: solid 1px undefined;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 24px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 4px;
  padding: 4px;
  width: 24px;
}

.c0 svg {
  height: 100%;
  width: 100%;
}

.c0:hover {
  opacity: 0.8;
}

<div>
    <button
      class="c0"
      color="secondary"
      name="name"
      type="submit"
    >
      <svg>
        close.svg
      </svg>
    </button>
  </div>
</body>
`;

exports[`IconButton component matches snapshots with additional props 2`] = `
<body>
  <div>
    <button
      class="sc-AxjAm DdLgh"
      color="primary"
      disabled=""
      name=""
      type="button"
    >
      <svg>
        close.svg
      </svg>
    </button>
  </div>
  <div>
    <button
      class="sc-AxjAm gvtLkn"
      color="secondary"
      name="name"
      type="submit"
    >
      <svg>
        close.svg
      </svg>
    </button>
  </div>
</body>
`;
